### PR TITLE
Manually cherry pick code introduced on #1548.

### PR DESCRIPTION
**Pull request by bordoni**
🎟 [TEC-3169]
---

[TEC-3169]: https://moderntribe.atlassian.net/browse/TEC-3169

### CODE REVIEW

File: src/views/blocks/rsvp/form/quantity-plus.php [hunk 1]
```diff
@@ -11,9 +11,13 @@
...
+ * @version TBD
  *
  */
 
 ?>
-<button class="tribe-block__rsvp__number-input-button tribe-block__rsvp__number-input-button--plus"></button>
+<button
+	type="button"
+	class="tribe-block__rsvp__number-input-button tribe-block__rsvp__number-input-button--plus"
+></button>
```
Comment (tr1b0t) [inline src/views/blocks/rsvp/form/quantity-plus.php]: :no_entry_sign: File must end with a newline character

<sub>hash: 21e4c68538601c54cb7c5348c85462c8</sub>

File: src/views/blocks/tickets/submit-button-modal.php [hunk 2]
```diff
@@ -21,20 +24,34 @@
...
 
 /**
- * Filter Modal Content
+ * Allow filtering of the button classes for the tickets block.
+ *
+ * @since TBD
+ *
+ * @param array $button_name The button classes.
+ */
+$button_classes = apply_filters( 'tribe_tickets_ticket_block_submit_classes', [
```
Comment (tr1b0t) [inline src/views/blocks/tickets/submit-button-modal.php]: * :no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call
* :no_entry_sign: Short array syntax is not allowed

<sub>hash: 1998f35cb83cce323dd67244227b70a9</sub>

File: src/views/blocks/tickets/submit-button-modal.php [hunk 3]
```diff
@@ -21,20 +24,34 @@
...
+	'tribe-common-c-btn',
+	'tribe-common-c-btn--small',
+	'tribe-tickets__buy'
```
Comment (tr1b0t) [inline src/views/blocks/tickets/submit-button-modal.php]: * :no_entry_sign: Comma required after last value in array declaration
* :no_entry_sign: Each array item in a multi-line array declaration must end in a comma

<sub>hash: 4464a4ef22b98e4f740b7b0b0edd9edc</sub>

File: src/views/blocks/tickets/submit-button-modal.php [hunk 4]
```diff
@@ -21,20 +24,34 @@
...
+] );
```
Comment (tr1b0t) [inline src/views/blocks/tickets/submit-button-modal.php]: :no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself

<sub>hash: 3fb479ee694a0e05eb10b4c4cac4e67f</sub>

File: src/views/blocks/tickets/submit-button.php [hunk 5]
```diff
@@ -23,14 +26,27 @@
...
 $button_name = apply_filters( 'tribe_tickets_ticket_block_submit', 'cart-button' );
+
+/**
+ * Allow filtering of the button classes for the tickets block.
+ *
+ * @since TBD
+ *
+ * @param array $button_name The button classes.
+ */
+$button_classes = apply_filters( 'tribe_tickets_ticket_block_submit_classes', [
```
Comment (tr1b0t) [inline src/views/blocks/tickets/submit-button.php]: * :no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call
* :no_entry_sign: Short array syntax is not allowed

<sub>hash: 72fb5f183b852113b377edde60fcdf8c</sub>

File: src/views/blocks/tickets/submit-button.php [hunk 6]
```diff
@@ -23,14 +26,27 @@
...
+	'tribe-common-c-btn',
+	'tribe-common-c-btn--small',
+	'tribe-tickets__buy',
+] );
```
Comment (tr1b0t) [inline src/views/blocks/tickets/submit-button.php]: :no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself

<sub>hash: fc3e50cd104e6896a2335379d7cf9bdc</sub>

File: src/views/blocks/tickets/submit-button.php [hunk 7]
```diff
@@ -23,14 +26,27 @@
...
 ?>
 <button
-	class="tribe-common-c-btn tribe-common-c-btn--small tribe-tickets__buy"
-	type="submit"
+	<?php tribe_classes( $button_classes ) ?>
```
Comment (tr1b0t) [inline src/views/blocks/tickets/submit-button.php]: :no_entry_sign: Inline PHP statement must end with a semicolon

<sub>hash: 8ee6c9f59c905a1a9000f293f4f2c91d</sub>

File: src/views/tickets/orders.php [hunk 8]
```diff
@@ -87,7 +88,14 @@
...
 	) : ?>
 		<div class="tribe-submit-tickets-form">
-			<button type="submit" name="process-tickets" value="1" class="button alt"><?php echo sprintf( esc_html__( 'Update %s', 'event-tickets' ), $view->get_description_rsvp_ticket( $event_id, get_current_user_id() ) ); ?></button>
+			<button
+				type="submit"
+				name="process-tickets"
+				value="1"
+				class="button alt"
+			>
+				<?php echo sprintf( esc_html__( 'Update %s', 'event-tickets' ), $view->get_description_rsvp_ticket( $event_id, get_current_user_id() ) ); ?>
```
Comment (tr1b0t) [inline src/views/tickets/orders.php]: * :no_entry_sign: A gettext call containing placeholders was found, but was not accompanied by a "translators:" comment on the line above to clarify the meaning of the placeholders.
* :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$view'
* :no_entry_sign: All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$view'.
* :no_entry_sign: All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found 'get_current_user_id'.
* :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')'

<sub>hash: 8b656697cd7a1974258a67d5a976ca67</sub>

File: src/views/tickets/tpp.php [hunk 9]
```diff
@@ -152,7 +154,12 @@ class="tribe-tickets-quantity qty"
 				</td>
 				<td class="tickets_submit">
 					<?php if ( ! $must_login ) : ?>
-						<button type="submit" class="tpp-submit tribe-button"><?php esc_html_e( 'Buy now', 'event-tickets' );?></button>
+						<button
+							type="submit"
+							class="tpp-submit tribe-button"
+						>
+							<?php esc_html_e( 'Buy now', 'event-tickets' );?>
```
Comment (tr1b0t) [inline src/views/tickets/tpp.php]: :no_entry_sign: Expected 1 space before closing PHP tag; 0 found

<sub>hash: f45489d40a64ec9863767df075c78406</sub>

File: src/views/blocks/tickets/quantity-remove.php [hunk 10]
```diff
@@ -11,7 +11,9 @@
  * @link {INSERT_ARTICLE_LINK_HERE}
  *
  * @since 4.9.3
- * @version 4.11.0
+ * @since TBD Updated the button to include a type - helps avoid submitting forms unintentionally.
+ *
+ * @version TBD
  *
  */
```
Comment (Camwyn) [inline src/views/blocks/tickets/quantity-remove.php]: This one is missing the change - let me check the current code though. It might be missing?

Comment (Camwyn) [reply src/views/blocks/tickets/quantity-remove.php]: It's in https://github.com/moderntribe/event-tickets/commit/b955ca810db1f1b93316e764d1f96a3809ee344f

File: src/views/blocks/tickets/submit-button-modal.php (hunk 2 above)
Comment (Camwyn) [reply src/views/blocks/tickets/submit-button-modal.php]: shut up bot.

File: src/views/blocks/tickets/quantity-remove.php (hunk 10 above)
Comment (bordoni) [reply src/views/blocks/tickets/quantity-remove.php]: Just added the change.

File: src/views/blocks/tickets/submit-button-modal.php [hunk 11]
```diff
@@ -21,20 +24,37 @@
...
- * Filter Modal Content
+ * Allow filtering of the button classes for the tickets block.
+ *
+ * @since TBD
+ *
+ * @param array $button_name The button classes.
+ */
+$button_classes = apply_filters(
+	'tribe_tickets_ticket_block_submit_classes',
+	[
```
Comment (tr1b0t) [inline src/views/blocks/tickets/submit-button-modal.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: af00d607c447e6cf7737bf3b4e74ba3f</sub>

File: src/views/blocks/tickets/submit-button.php [hunk 12]
```diff
@@ -23,14 +26,30 @@
...
+/**
+ * Allow filtering of the button classes for the tickets block.
+ *
+ * @since TBD
+ *
+ * @param array $button_name The button classes.
+ */
+$button_classes = apply_filters(
+	'tribe_tickets_ticket_block_submit_classes',
+	[
```
Comment (tr1b0t) [inline src/views/blocks/tickets/submit-button.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: 3e25065ee413a465ebce8d72fdc7c357</sub>

File: src/views/tickets/orders.php (hunk 8 above)
Comment (tr1b0t) [inline src/views/tickets/orders.php]: * :no_entry_sign: A gettext call containing placeholders was found, but was not accompanied by a "translators:" comment on the line above to clarify the meaning of the placeholders.
* :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$view'
* :no_entry_sign: All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$view'.
* :no_entry_sign: All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found 'get_current_user_id'.
* :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')'

<sub>hash: 8b656697cd7a1974258a67d5a976ca67</sub>

File: src/views/tickets/tpp.php (hunk 9 above)
Comment (tr1b0t) [inline src/views/tickets/tpp.php]: :no_entry_sign: Expected 1 space before closing PHP tag; 0 found

<sub>hash: f45489d40a64ec9863767df075c78406</sub>

**Pull request merged**